flake.nix

```diff
@@ -110,7 +110,31 @@
             };
           };
 
-          terraformProviders = pkgs.terraform-providers.actualProviders;
+          terraformProviders = let
+            terraform-providers = pkgs.terraform-providers.actualProviders;
+          in
+            lib.mapAttrs
+            (
+              _: provider:
+                if provider ? override
+                then
+                  # use opentofu plugin registry over terraform's
+                  provider.override (oldArgs: {
+                    provider-source-address =
+                      lib.replaceStrings ["https://registry.terraform.io/providers"] [
+                        "registry.opentofu.org"
```

Comment:
Maybe stupid question, but why do the opentofu version can (our should) do without the `https://`?

Reply:
nah good question, i don't actually know the answer. i'll try and add that in the upstream (and factor the patch out over here).

Reply:
or rather, @yajo i think i originally got this patch from [your comment](https://github.com/NixOS/nixpkgs/issues/283015#issuecomment-2009610478), would you maybe recall why you'd dropped the scheme for opentofu?

Reply:
i see. whatever's causing this, leaving the scheme in seems to make it fail, testing on an altered nixpkgs opentofu package:

```
$ nix-build -A opentofu.tests.opentofu_plugins_test --show-trace
this derivation will be built:
  /nix/store/sf22cpzmq61csnmkjgcrbks06gh4gc1f-opentofu-plugin-test.drv
building '/nix/store/sf22cpzmq61csnmkjgcrbks06gh4gc1f-opentofu-plugin-test.drv'...

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/random...
╷
│ Error: Failed to query available provider packages
│
│ Could not retrieve the list of available versions for provider hashicorp/random: provider registry.opentofu.org/hashicorp/random was not found in any of the search locations
│
│   - /nix/store/fd09rv0ra862wrkmppczb4ddmd0pmgxb-opentofu-1.8.7/libexec/terraform-providers
╵

error: builder for '/nix/store/sf22cpzmq61csnmkjgcrbks06gh4gc1f-opentofu-plugin-test.drv' failed with exit code 1;
       last 13 log lines:
       >
       > Initializing the backend...
       >
       > Initializing provider plugins...
       > - Finding latest version of hashicorp/random...
       > ╷
       > │ Error: Failed to query available provider packages
       > │
       > │ Could not retrieve the list of available versions for provider hashicorp/random: provider registry.opentofu.org/hashicorp/random was not found in any of the search locations
       > │
       > │   - /nix/store/fd09rv0ra862wrkmppczb4ddmd0pmgxb-opentofu-1.8.7/libexec/terraform-providers
       > ╵
       >
       For full logs, run 'nix log /nix/store/sf22cpzmq61csnmkjgcrbks06gh4gc1f-opentofu-plugin-test.drv'.
```

Reply:
Well, that's a pretty good reason to keep it this way then - maybe leaving a honest comment if someone else wonder in the future (we don't really know why, but if we had the scheme, it doesn't work anymore)

Reply:
I cannot remember the reason, but it works.